tracks/julia/exercises/word-count/mentoring.md

```diff
@@ -0,0 +1,124 @@
+# Example solutions
+
+`````
+
+Docstring:
+
+````julia
+"""
+    wordcount(input)
+
+Find the frequency of each term in the input.
+
+## Example
+
+```jldoctest
+julia> wordcount("Joe can't tell between 'large' and large.")
+Dict{String,Int64} with 6 entries:
+  "tell"    => 1
+  "can't"   => 1
+  "large"   => 2
+  "joe"     => 1
+  "between" => 1
+  "and"     => 1
+```
+"""
+````
+
+**Split, then strip quotation**
+
+This solution, like all the others below uses "Regex".
+Regex is a concise language for describing patterns in strings.
+
+In Julia, `r"foo"` constructs a Regex at compilation time using the `@r_str` macro.
+
+```julia
+function wordcount(input)
+    words = split(input, r"[^\w']+")
+    counts = Dict{String, Int}()
+    for word in words
+        # Removing leading and closing apostrophes (might be quotation)
+        word = lowercase(strip(word, '''))
+        if !isempty(word)
+            counts[word] = get(counts, word, 0) + 1
+        end
+    end
+    return counts
+end
+```
+
+This solution uses a regex, but you could equally do something like:
+
+```julia
+split(input, c -> !(isalnum(c) || c == '''); keepempty=false)
```

Comment:
Where are you getting the `isalnum` function?

Also this isn't quite right, because splitting on `'` is not desired for a contraction. This solution (which of course can be `countmap`-ed manually if desired) splits only on `isspace`, and then strips punctuation separately (only removing punct chars on the edges of words). Filtering for empty strings has to happen after that anyway in case an empty string was formed by stripping punctuation.

```julia
wordcount(sent) = countmap(filter!(!isempty, strip.(ispunct, split(lowercase(sent)))))
```

Reply:
Seems like `isalnum` has been deprecated, which I wasn't aware of: https://discourse.julialang.org/t/isalnum-deprecated/8504

Reply:
Yeah, I should have checked that.

I think `isletter(c) || isdigit(c) || c == '''` should be used.

The idea is to replicate the regex pretty closely, so I don't want to show your method here.

Should maybe show it later, tho? Would need to introduce `countmap`, too.

Reply:
The regex also includes underscore, which isn't desirable. Maybe we should mention that